test/powershell/engine/Help/UpdatableHelpSystem.Tests.ps1

```diff
@@ -0,0 +1,268 @@
+$expectedCompressedFiles["Microsoft.PowerShell.Archive"] = @("Microsoft.PowerShell.Archive_eb74e8da-9ae2-482a-a648-e96550fb8733_en-US_HelpContent$extension")
+$expectedCompressedFiles["Microsoft.PowerShell.Core"] = @("Microsoft.PowerShell.Core_00000000-0000-0000-0000-000000000000_en-US_HelpContent$extension")
+$expectedCompressedFiles["Microsoft.PowerShell.Diagnostics"] = @("Microsoft.PowerShell.Diagnostics_ca046f10-ca64-4740-8ff9-2565dba61a4f_en-US_helpcontent$extension")
+$expectedCompressedFiles["Microsoft.PowerShell.Host"] = @("Microsoft.PowerShell.Host_56d66100-99a0-4ffc-a12d-eee9a6718aef_en-US_helpcontent$extension")
+$expectedCompressedFiles["Microsoft.PowerShell.LocalAccounts"] = @("Microsoft.PowerShell.LocalAccounts_8e362604-2c0b-448f-a414-a6a690a644e2_en-US_HelpContent$extension")
+$expectedCompressedFiles["Microsoft.PowerShell.Management"] = @("Microsoft.PowerShell.Management_eefcb906-b326-4e99-9f54-8b4bb6ef3c6d_en-US_helpcontent$extension")
+$expectedCompressedFiles["Microsoft.PowerShell.Security"] = @("Microsoft.PowerShell.Security_a94c8c7e-9810-47c0-b8af-65089c13a35a_en-US_helpcontent$extension")
+$expectedCompressedFiles["Microsoft.PowerShell.Utility"] = @("Microsoft.PowerShell.Utility_1da87e53-152b-403e-98dc-74d7b4d63d59_en-US_helpcontent$extension")
+$expectedCompressedFiles["Microsoft.WsMan.Management"] = @("Microsoft.WsMan.Management_766204A6-330E-4263-A7AB-46C87AFC366C_en-US_helpcontent$extension")
+$expectedCompressedFiles["PackageManagement"] = @("PackageManagement_4ae9fd46-338a-459c-8186-07f910774cb8_en-US_helpcontent$extension")
+$expectedCompressedFiles["PowershellGet"] = @("PowershellGet_1d73a601-4a6c-43c5-ba3f-619b18bbb404_en-US_helpcontent$extension")
+$expectedCompressedFiles["PSReadline"] = @("PSReadline_5714753b-2afd-4492-a5fd-01d9e2cff8b5_en-US_helpcontent$extension")
+
+# These are the paths where the help content will be installed to.
+$getHelpContentPaths = @(
+    "$pshome\en-US"
+    "$pshome\Modules\CimCmdlets\en-US"
+    "$pshome\Modules\Microsoft.PowerShell.Archive\en-US"
+    "$pshome\Modules\Microsoft.PowerShell.LocalAccounts\en-US"
+    "$pshome\Modules\PackageManagement\en-US"
+    "$pshome\Modules\PowershellGet\en-US"
+    "$pshome\Modules\PSReadline\en-US"
+)
+    
+function DeleteInboxHelpContent
+{
+    foreach ($path in $getHelpContentPaths)
+    {
+        if (Test-Path $path)
+        {
+            Get-ChildItem $path -Include @("about_*.txt","*help.xml") -Recurse -ea SilentlyContinue | Remove-Item -Force -ErrorAction SilentlyContinue
+        }
+    }
+}
+
+function GetFiles
+{
+    param (
+        [string]$fileType = "*help.xml",
+        [string[]]$path = $getHelpContentPaths
+    )
+                            
+    foreach ($thisPath in $path)
+    {
+        if (Test-Path $thisPath)
+        {
+            return @(Get-ChildItem $thisPath -Include $fileType -Recurse -ea SilentlyContinue | foreach {$_.FullName})
+        }
+    }
```

Comment:
This can be replaced with Get-ChildItem $path -Include $fileType -Recurse -ea SilentlyContinue | Select-Object -ExpandProperty FullName

Since it already has -ea Silentlycontinue, test-path is not required as the error will be ignored anyways. Using select-object we avoid having 2 foreach loops.